db/lastfm_session.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+import os
+import binascii
+import db
+from sqlalchemy import text
+from lastfm_user import User
+
+
+class Session(object):
+    """ Session class required by the api-compat """
+
+    def __init__(self, row):
+        self.id, userid, self.sid, self.api_key, self.timestamp = row
+        self.user = User.load_by_id(userid)
+
+    @staticmethod
+    def load(session, api_key=None):
+        """ Load the session details from the database.
+            If the api_key is also supplied then verify it as well.
+        """
+        params = {'sid': session}
+        query = "SELECT * FROM session WHERE sid=:sid"
+        if api_key:
+            params['api_key'] = api_key
+            query = "SELECT * FROM session WHERE sid=:sid AND api_key=:api_key"
+
+        with db.engine.connect() as connection:
+            result = connection.execute(text(query), params)
+            row = result.fetchone()
+            if row:
+                return Session(row)
```

Comment:
There's quite a bit of "magic" in this line. In the select query you get `*` (all columns), and in the constructor for `Session`, it expects that there are 5 values (`self.id, userid, self.sid, self.api_key, self.timestamp = row`). This means that if we ever add/remove a column to this table, and don't update this or have tests, we're going to have a problem.

In AcousticBrainz we treat results of `connection.execute` as a dictionary. In fact, you can do this:
```python
res = connection.execute(text(query), params)
return dict(res)
```
and you'll get a dictionary back.

Our preference for SQL statements is to explicitly list all columns that we want, for example see here: https://github.com/metabrainz/acousticbrainz-server/blob/master/db/dataset.py#L120

My preference would be for these queries to explicitly select the columns that we need, and then explicitly pass 5 parameters into the `Session` object by name.